commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java

```diff
@@ -91,7 +91,13 @@ protected SftpFileSystem(final GenericFileName rootName, final Session session,
         this.session = Objects.requireNonNull(session, "session");
         this.connectTimeoutMillis = SftpFileSystemConfigBuilder.getInstance()
                 .getConnectTimeoutMillis(fileSystemOptions);
-        this.execDisabled = detectExecDisabled();
+        
+        if (SftpFileSystemConfigBuilder.getInstance().isExecEnabled(fileSystemOptions)){
+            this.execDisabled = detectExecDisabled();

```

Comment:
This feature disables _detection_ of the exec channel, so it should be called as such IMO: `EXEC_ENABLED` -> `DISABLE_DETECT_EXEC_CHANEL`.